File: ui-admin/src/app/backend/criteria/business/stock-movement-criteria.ts
import { CommonCriteria } from '../commoncriteria';

export class StockMovementCriteria extends CommonCriteria {
    id: number;
    productId: number;
    categoryId: number;
    posId: number;
    movementType: string;
    reason: string;

    constructor() {
        super();
        this.id = null;
        this.productId = null;
        this.categoryId = null;
        this.posId = null;
        this.movementType = null;
        this.reason = null;
    }
}

File: api/src/main/java/com/mystore/manager/api/business/common/criteria/StockMovementCriteria.java
package com.mystore.manager.api.business.common.criteria;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.mystore.manager.api.common.criteria.PaginationCriteria;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@JsonIgnoreProperties(ignoreUnknown = true)
public class StockMovementCriteria extends PaginationCriteria {
    private Integer id;
    private Integer productId;
    private Integer categoryId;
    private Integer posId;
    private String movementType;
    private String reason;
}
